### ISSUE TITLE
In-place restores should shutdown the cluster by default

### ISSUE THREAD

**Issue by jsanda** [k8ssandra/k8ssandra#611]
**Is your feature request related to a problem? Please describe.**
In-place restores are done via rolling restart by default. If the cluster remains available and continues accepting writes while the restore is in process, we cannot guarantee that the cluster will restored to the backup state. 

Furthermore, there is a possibility that writes could be lost. During the restore process Medusa deletes the contents of the data subdirectory of the table being restored. This could easily lead to data loss.

**Describe the solution you'd like**
Shut down the entire cluster by default to be proactive in preventing data loss or inconsistent state during a restore.

It is worth noting that shutting down the cluster won't completely prevent these problems. As nodes come back online, clients could still perform queries before the restore is complete. Completely preventing client access is beyond the scope of this issue and might be something we can better address when we do service mesh integration.

**Describe alternatives you've considered**

**Additional context**
Restores are currently are really, really slow. Part of that is due to #570 and #571. 

The other reason is that the restore operation is done serially one Cassandra pod at a time. The restore operation is performed by the `medusa-restore` init container. The bulk of the time of the restore operation will be from downloading the backup files. Doing it serially one node at a time will make it painfully slow and could result in result in long periods of downtime for the cluster. 

The reason it is done serially is because the restore happens by way of a StatefulSet update. The StatefulSet controller does a rolling update of pods, one at a time. I think we can work around this relatively easily. The StatefulSet's `.spec.podManagementPolicy` property controls how pods are created. The default value is `OrderedReady` which means one pod at a time. cass-operator uses `Parallel`. If we simply delete the StatefulSet(s) and let them get recreated, the `medusa-restore` init containers will run in parallel. Note that deleting a Statefulset does not cascade to the PersistentVolumeClaims and PersistentVolumes that it uses.

**Comment by burmanm** [comment]
This isn't too complex, we already have CassandraBackup property "Shutdown" which will shutdown the cluster during in-place restore. We simply negate the behavior to make it the default process.

**Comment by jsanda** [comment]
The way the docs are written I was under the impression that StatefulSet scale up (after initial creation) happens sequentially even when using a `PodManagementPolicy` of `Parallel`. I tested some more and verified that subsequent scale up do happen in parallel which makes things a lot simpler.

>This isn't too complex, we already have CassandraBackup property "Shutdown" which will shutdown the cluster during in-place restore. We simply negate the behavior to make it the default process.

I am not a fan of this. It will be difficult to differentiate between `Shutdown` being unset (in which case we take the inverse) versus the user actually wanting a value of `false`. This is why I started the changes to make `Shutdown` a pointer.

For the upcoming k8ssandra 1.1 release, it will be sufficient to just change the default value of the `shutdown` property of the k8ssandra/restore chart.
